v22.2/backup-validation.md

```diff
@@ -0,0 +1,143 @@
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SHOW TABLES FROM movr;
+~~~
+~~~
+  schema_name |         table_name         | type  | owner | estimated_row_count | locality
+--------------+----------------------------+-------+-------+---------------------+-----------
+  public      | promo_codes                | table | root  |                   0 | NULL
+  public      | rides                      | table | root  |                   0 | NULL
+  public      | user_promo_codes           | table | root  |                   0 | NULL
+  public      | users                      | table | root  |                   0 | NULL
+  public      | vehicle_location_histories | table | root  |                   0 | NULL
+  public      | vehicles                   | table | root  |                   0 | NULL
+(6 rows)
+~~~
+
+You'll find the tables in place with no user table data.
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SELECT * FROM rides;
+~~~
+~~~
+  id | city | vehicle_city | rider_id | vehicle_id | start_address | end_address | start_time | end_time | revenue
+-----+------+--------------+----------+------------+---------------+-------------+------------+----------+----------
+(0 rows)
+~~~
+
+Once you have successfully validated the restore, you can revert the cluster to its pre-restore state by dropping the table or database:
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+DROP DATABASE movr CASCADE;
+~~~
+
+### Cluster-level backup validation
+
+It is important to note that [full cluster restores](restore.html#full-cluster) with `schema_only` will write the system tables to disk. This provides important coverage for validation at the cluster level. Writing the system tables should not have a notable impact on the runtime of this process.
+
+Once you have successfully validated a cluster-level restore, the restore system data cannot be reverted. However, you can drop the databases and tables as per the previous command.
+
+## Validate the restore of backup table data
+
+A restore with the `verify_backup_table_data` option will perform a `schema_only` restore and the following:
+
+1. Read all user data from the storage location. 
+1. Checksum the user data to ensure it is not corrupt.
+1. Discard the user data before it is written to disk.
+
+In comparison to `schema_only` restores, `verify_backup_table_data` also validates the backup by reading and checksumming the user data.
+
+It is necessary to include `schema_only` when you run a restore with `verify_backup_table_data`:
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+RESTORE DATABASE movr FROM LATEST IN "s3://bucket?AWS_ACCESS_KEY_ID={Access Key ID}&AWS_SECRET_ACCESS_KEY={Secret Access Key}" WITH schema_only, verify_backup_table_data;
+~~~
+
+Similarly, to just `schema_only` restores, you'll find the table schemas restored. You'll also receive estimated row counts as the restore reads the files during the process: 
```

Comment:
I added this sentence re the estimated row counts. The estimated_row_counts come from the system statistics table, I think, so not sure if this is wholly useful here. Let me know.

Reply:
wait, I thought my[ fix](https://github.com/cockroachdb/cockroach/pull/88045) caused the estimated_row_count to be zero for any schema_only restore?

Reply:
I don't we need to talk about estimated row counts in the docs, as they should be 0 across the board.

Reply:
So, I ran these examples on 22.1.0-alpha.3 and got estimated_row_count 0 across the tables for `schema_only` alone. But, when running schema_only + verify_backup_table_data, I received the estimations as copied here.

Reply:
ah, that release doesn't have my fix. It looks like you updated the docs page to show that row sizes are 0.

Reply:
I have taken out that table on the `verify_table_data` section altogether since it shows the same thing as the `schema_only` section.

Reply:
great!